### PR TITLE
Add mysql2 support

### PR DESCRIPTION
**Pull request by prog-supdex**
## What is the purpose of this pull request?
Added mysql2 support

## Is there anything you'd like reviewers to focus on?
Added a new Mysql Adapter
Mysql doesn't support `ON CONFLICT` (but has `ON DUPLICATE KEY UPDATE`) and doesn't have `RETURNING ID`

So, I used `ActiveRecord::Base.connection.update(sql)` for getting amount of updated rows
Also, Mysql doesn't understand `EXCLUDED`, so we need to use some method `wrap_column_name`, which will be defined in every adapter. Because, Mysql understands `VALUES(here_column_names)`, but Postgresql understands `EXCLUDED.here_column_name`

## Checklist

- [- ] I've added tests for this change
- [+] I've added a Changelog entry
- [+] I've updated a documentation

### CODE REVIEW

File: lib/activerecord_slotted_counters/adapters/mysql_upsert.rb [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+module ActiveRecordSlottedCounters
+  module Adapters
+    class MysqlUpsert
+      attr_reader :klass, :current_adapter_name
+
+      def initialize(klass, current_adapter_name)
+        @klass = klass
+        @current_adapter_name = current_adapter_name
```
Comment (palkan) [inline lib/activerecord_slotted_counters/adapters/mysql_upsert.rb:10]: We don't this to be a part of the adapter state; we only use it in the `apply?` method, so let's pass current adapter as an argument to `#apply?` instead.

File: lib/activerecord_slotted_counters/adapters/pg_upsert.rb [hunk 2]
```diff
@@ -3,14 +3,18 @@
 module ActiveRecordSlottedCounters
   module Adapters
     class PgUpsert
-      attr_reader :klass
+      attr_reader :klass, :current_adapter_name
 
-      def initialize(klass)
+      def initialize(klass, current_adapter_name)
         @klass = klass
+        @current_adapter_name = current_adapter_name
```
Comment (palkan) [inline lib/activerecord_slotted_counters/adapters/pg_upsert.rb:10]: Ditto

**Review by palkan — CHANGES_REQUESTED** [review]

File: lib/activerecord_slotted_counters/adapters/mysql_upsert.rb (hunk 1 above)
Comment (prog-supdex) [reply lib/activerecord_slotted_counters/adapters/mysql_upsert.rb]: thanks
got it!
done!

File: lib/activerecord_slotted_counters/adapters/pg_upsert.rb (hunk 2 above)
Comment (prog-supdex) [reply lib/activerecord_slotted_counters/adapters/pg_upsert.rb]: fixed

**Review by palkan — APPROVED** [review]

**Comment by palkan** [comment]
Thanks!

**Pull request merged**